bayesian/utils.py

```diff
@@ -455,3 +455,79 @@ def get_files_github_url(self, github_url):
+        :param manifest_response: dict. object having all recommendation elements
+        :return: same dict. object with populated reasons
+        """
+        # Populate reason for each companion package
+        manifest_response = self._companion_reason(manifest_response)
+
+        # Populate reason for each alternate package
+        manifest_response = self._alternate_reason(manifest_response)
+        return manifest_response
+
+    def _alternate_reason(self, manifest_response):
+        """
+        It will populate simple reason for each alternate package recommendation
+        :param manifest_response: dict. object having all recommendation elements
+        :return: same dict. object with populated reasons for alternate package
+        """
+        length = len(manifest_response[0].get("recommendation").get("alternate"))
+        if length > 0:
+            count = 0
+            while count < length:
+                test_usage_outlier = self._check_usage_outlier(count, manifest_response)
+                sentence = ""
+                if test_usage_outlier:
+                    sentence = manifest_response[0].get("recommendation").get("alternate")[count].get("name") + \
+                               " is recommended as an alternate package for " + \
+                               manifest_response[0].get("recommendation").get("alternate")[0].get("replaces")[
+                                   0].get("name") \
+                               + " as very few stacks are using it with provided input stack."
+                manifest_response[0]["recommendation"]["alternate"][count]["reason"] = sentence
+                count = count + 1
+        return manifest_response
+
+    def _check_usage_outlier(self, count, manifest_response):
+        """
+        It will check usage outlier a package mentioned by count index
+        :param count: Index of input package in manifest_response
+        :param manifest_response: dict. object having all recommendation elements
+        :return: True or False for usage outlier of input package at count index position
+        """
+        test = False
+        pkg_name = manifest_response[0].get("recommendation").get("alternate")[0].get("replaces")[count].get("name")
+        outliers = manifest_response[0].get("recommendation").get("usage_outliers")
+        for outlier in outliers:
+            if pkg_name == outlier.get("package_name"):
+                test = True
+                break
+        return test
+
+    def _companion_reason(self, manifest_response):
+        """
+        It will populate the simple reason for each companion package
+        :param manifest_response: dict. object having all recommendation elements
+        :return: same dict. object with populated reasons for each companion package
+        """
+        length = len(manifest_response[0].get("recommendation").get("companion"))
+        if length > 0:
+            count = 0
+            while count < length:
```

Comment:
same comment as above. If you can implement in more pythonic way

Reply:
Done